icon/chainscore_iiss.go

```diff
@@ -671,15 +671,20 @@ func (s *chainScore) Ex_setRewardFund(iglobal *big.Int) error {
 	revision := s.cc.Revision().Value()
 	if revision <= icmodule.RevisionIISS4R0 {
 		rf := es.State.GetRewardFundV1()
-		rf.SetIGlobal(iglobal)
+		if err = rf.SetIGlobal(iglobal); err != nil {
+			return err
+		}
 		if err = es.State.SetRewardFund(rf); err != nil {
 			return err
 		}
 	}
 
 	if revision >= icmodule.RevisionIISS4R0 {
+		iiss.EmitRewardFundSetEvent(s.newCallContext(s.cc), iglobal)
 		rf := es.State.GetRewardFundV2()
-		rf.SetIGlobal(iglobal)
+		if err = rf.SetIGlobal(iglobal); err != nil {
+			return err
+		}
 		return es.State.SetRewardFund(rf)
```

Comment:
Just a question:
Don't we need to care about function call order between emitting an event and state update?

Reply:
No need to care about that.
For failed TX, events will not be stored and step will not be charged for it.